components/Login.tsx

```diff
@@ -0,0 +1,89 @@
+import React from "react";
+import { useForm } from "react-hook-form";
+import { yupResolver } from "@hookform/resolvers/yup";
+import * as Yup from "yup";
+import Link from "next/link";
+import { ILoginInput } from "@/types/types";
+import ErrorMessage from "./Common/ErrorMessage";
+import { logIn } from "@/store/firebaseHelpers";
+
+const Login = () => {
+  // form validation
+  const validationSchema = Yup.object().shape({
+    email: Yup.string().required("Email is required").email("Email is invalid"),
+    password: Yup.string().required("Password is required"),
+  });
+  const formOptions = { resolver: yupResolver(validationSchema) };
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<ILoginInput>(formOptions);
+
+  // onSubmit handler
+  const onSubmit = async (data: ILoginInput) => {
+    try {
+      await logIn(data);
+    } catch (error: any) {
```

Comment:
No `any` 
Also show a toast or something if this fails - handling error cases comprehensively is a cornerstone of good UX